File: app/src/main/java/com/example/myapplication/view/activity/DetailDogActivity.java
package com.example.myapplication.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.R;
import com.example.myapplication.adapter.DetailDogAdapter;
import com.example.myapplication.data.model.DogDetailModel;
import com.example.myapplication.databinding.ActivityDetaildogBinding;
import com.example.myapplication.view.custom.BaseActivity;

public class DetailDogActivity extends BaseActivity {

    public static final String TAG_DOG_ENTITY = "dogEntity";


    private ActivityDetaildogBinding binding;
    private DetailDogAdapter adapter;
    private DogDetailModel model;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_detaildog);
        init();
    }

    public void getIntentData() {
        Intent intent = getIntent();
        model.setInfo(intent.getParcelableExtra(TAG_DOG_ENTITY));
    }

    @Override
    public void init() {
        model = new DogDetailModel();
        getIntentData();

        adapter = new DetailDogAdapter(this, model);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(RecyclerView.VERTICAL);
        binding.recyclerview.setLayoutManager(linearLayoutManager);
        binding.recyclerview.setAdapter(adapter);
        setListener();
    }

    @Override
    public void setListener() {

    }

    @Override
    public void onClick(View view) {

    }
}

File: app_kotlin/src/main/java/com/example/myapplication/view/activity/DetailDogActivity.kt
package com.example.myapplication.view.activity

import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.example.myapplication.R
import com.example.myapplication.adapter.DetailDogAdapter
import com.example.myapplication.data.model.DogDetailModel
import com.example.myapplication.databinding.ActivityDetaildogBinding
import com.example.myapplication.view.custom.BaseActivity

class DetailDogActivity : BaseActivity() {


    private var binding: ActivityDetaildogBinding? = null
    private var adapter: DetailDogAdapter? = null
    private var model: DogDetailModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_detaildog)
        init()
    }

    fun getIntentData() {
        val intent = intent
        model!!.info = intent.getParcelableExtra(TAG_DOG_ENTITY)
    }

    override fun init() {
        model = DogDetailModel()
        getIntentData()

        adapter = DetailDogAdapter(this, model!!)
        val linearLayoutManager = LinearLayoutManager(this)
        linearLayoutManager.orientation = RecyclerView.VERTICAL
        binding!!.recyclerview.layoutManager = linearLayoutManager
        binding!!.recyclerview.adapter = adapter
        setListener()
    }

    override fun setListener() {

    }

    override fun onClick(view: View) {

    }

    companion object {
        const val TAG_DOG_ENTITY = "dogEntity"
    }
}
